lib/live/form.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require_relative 'view'
+require 'trenni/builder'
+require 'json'
+
+module Live 
+    #Represents a Form within the view layer
+    #Differs from Live::View because the server sends both the 
+    #details and a serialized form from the view layer.
+    class Form < Live::View 
+        
+        #returns [String] javascript code to execute when the form is submitted.
+        def handleForm(details=false)
+            if details 
+                return "live.handleForm(this.id, event, #{details})"
+            else 
+                return "live.handleForm(this.id, event)"
+            end 
+        end 
+
+        #Render the element
+        #returns [Object] Renders the given builder inside a form tag.
+        def to_html()
+
+            Trenni::Builder.fragment do |builder|
+                builder.tag :form, method: "post", id: @id, class: 'live', onsubmit: handleForm, data: @data do
+                    render(builder)
+                end
+            end
+        end 
+
+        #Handles an incoming event.
+        # @parameter load [String] the parsed message from the view layer, 
+        # which includes the details and the serialized form data.
+        def handle(event, message)
```

Comment:
I believe you should introduce a new method above called `def submit(form_data)`

This method should then invoke that with the argument rather than setting an instance variable.

The sub-class can be responsible for updating data and/or form data. Maybe we should repurpose `@data` - i.e. fields that are stored in the form are serialized as part of the form itself anyway, so it doesn't need to be part of the `data-` attributes. Not sure best way.